examples/2d/frustum_sprites.rs

```diff
@@ -0,0 +1,83 @@
+use bevy::{
+    diagnostic::{FrameTimeDiagnosticsPlugin, LogDiagnosticsPlugin},
+    prelude::*,
+};
+
+const CAMERA_SPEED: f32 = 1000.0;
+
+pub struct CameraTimer(Timer);
+pub struct Direction(Vec3);
+
+fn main() {
+    App::build()
+        .insert_resource(ReportExecutionOrderAmbiguities)
+        .add_plugin(LogDiagnosticsPlugin::default())
+        .add_plugin(FrameTimeDiagnosticsPlugin::default())
+        .insert_resource(bevy::log::LogSettings {
+            level: bevy::log::Level::DEBUG,
+            ..Default::default()
+        })
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(move_camera.system())
+        .run()
+}
+
+fn setup(
+    commands: &mut Commands,
+    assets: Res<AssetServer>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    let tile_size = Vec2::new(64.0, 64.0);
+    let map_size = Vec2::new(320.0, 320.0);
+
+    let half_x = (map_size.x / 2.0) as i32;
+    let half_y = (map_size.y / 2.0) as i32;
+
+    let material = materials.add(assets.load("textures/icon.png").into());
+
+    commands
+        .spawn(OrthographicCameraBundle::new_2d())
+        .with(CameraTimer(Timer::from_seconds(5.0, true)))
+        .with(Direction(Vec3::new(1.0, 1.0, 0.0)));
+
+    for y in -half_y..half_y {
+        for x in -half_x..half_x {
+            let position = Vec2::new(x as f32, y as f32);
+            let translation = (position * tile_size).extend(0.0);
+
+            commands.spawn(SpriteBundle {
+                material: material.clone(),
+                transform: Transform::from_translation(translation),
+                sprite: Sprite {
+                    size: tile_size,
+                    resize_mode: SpriteResizeMode::Manual,
+                },
+                ..Default::default()
+            });
+        }
+    }
+}
+
+pub fn move_camera(
+    time: Res<Time>,
+    sprites: Query<&Sprite>,
+    visible: Query<&Sprite, With<Visible>>,
+    mut query: Query<(&mut Transform, &mut Direction, &mut CameraTimer)>,
+) {
+    for (mut transform, mut direction, mut timer) in query.iter_mut() {
```

Comment:
Technically, these should probably be multiple queries/systems since the direction swapping is completely unrelated to swapping the direction

That being said, in this case it's fine.